pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/notPepperKit/NotPepperKitCreationService.java

```diff
@@ -0,0 +1,238 @@
+package org.broadinstitute.dsm.model.notPepperKit;
+
+import com.easypost.exception.EasyPostException;
+import com.easypost.model.Address;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.ddp.db.TransactionWrapper;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.db.KitRequestShipping;
+import org.broadinstitute.dsm.db.dao.ddp.instance.DDPInstanceDao;
+import org.broadinstitute.dsm.db.dto.ddp.instance.DDPInstanceDto;
+import org.broadinstitute.dsm.model.KitRequest;
+import org.broadinstitute.dsm.model.KitRequestSettings;
+import org.broadinstitute.dsm.model.KitType;
+import org.broadinstitute.dsm.util.DDPKitRequest;
+import org.broadinstitute.dsm.util.EasyPostUtil;
+import org.broadinstitute.lddp.util.DeliveryAddress;
+
+import java.sql.Connection;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+@Slf4j
+public class NotPepperKitCreationService {
+
+    public final static String ADDRESS_VALIDATION_ERROR = "UNABLE_TO_VERIFY_ADDRESS";
+    public final static String UNKNOWN_KIT_TYPE = "UNKNOWN_KIT_TYPE";
+    public final static String UNKNOWN_STUDY = "UNKNOWN_STUDY";
+    public final static String MISSING_JUNIPER_KIT_ID ="MISSING_JUNIPER_KIT_ID";
+    public final static String MISSING_JUNIPER_PARTICIPANT_ID ="MISSING_JUNIPER_PARTICIPANT_ID";
+    public final static String JUNIPER ="JUNIPER";
+    public final static String JUNIPER_UNDERSCORE ="JUNIPER_";
+
+    public KitResponse createNonPepperKit(JuniperKitRequest juniperKitRequest, String studyGuid, String kitTypeName) {
+        if (StringUtils.isBlank(juniperKitRequest.getJuniperParticipantID())) {
+            return new KitResponse(MISSING_JUNIPER_PARTICIPANT_ID, juniperKitRequest.getJuniperKitId());
+        }
+        if (StringUtils.isBlank(juniperKitRequest.getJuniperKitId())) {
+            return new KitResponse(MISSING_JUNIPER_KIT_ID, null);
+        }
+        Optional<DDPInstanceDto> maybeDdpInstanceDto = new DDPInstanceDao().getDDPInstanceByStudyGuid(studyGuid);
+        if (maybeDdpInstanceDto.isEmpty()) {
+            return new KitResponse(UNKNOWN_STUDY, juniperKitRequest.getJuniperKitId());
+        }
+        DDPInstanceDto ddpInstanceDto = maybeDdpInstanceDto.get();
+        //getting the instance with isHasRole being set to true if the instance has role juniper_study
+        DDPInstance ddpInstance = DDPInstance.getDDPInstanceWithRole(ddpInstanceDto.getInstanceName(), "juniper_study");
+
+        if (!ddpInstance.isHasRole()) {
```

Comment:
Are we leaking information by not just saying this is an invalid study ID, as above?
IOW, I'm not sure this provides useful information to a legitimate user, but would provide information to a bad actor.